### ISSUE TITLE
Questions about NLL computation in the "maze2d" branch

### ISSUE THREAD

**Issue by Jang-seunghwan** [Weixy21/SafeDiffuser#9]
While attempting to compute the negative log-likelihood (NLL) for the Maze2D task, 
I ran into a few points that I’m hoping you can clarify:

1. L_0 term in `vb_terms_bpd`
The current implementation comments out the L_0 calculation
- Is it correct to simply uncomment this block when we want to include L_0 in the NLL?
- I assume the L_T term was omitted in function because its contribution is ~0; is it right?

2. Potential mismatch in the ELBO call
In diffuser/guides/policies.py the ELBO is computed with
`elbo_i = self.diffusion_model._vb_terms_bpd(
    diffuser_state, conditions, diffusion[:, step, :, :], timesteps
)
`
Here, `diffusion[:, step, :, :]` seems to be an x_t sampled from an independent diffusion trajectory, 
so it may not align with `diffuser_state`. Is this intentional, or should both tensors come from the same trajectory?

3. Division NLL by 255 
`sum_elbo = sum_elbo.detach().cpu().numpy()[0] / 255  # ave`
It makes sense for image data, where pixel values lie in 0…255. But the Maze2D trajectories are already scaled to the range −1,1, 
so this extra division seems unnecessary.
Could you share the reasoning behind this 255-scaling step?

**Comment by Weixy21** [comment]
1. Yes, you can uncomment this block if you want to include L0

2. The diffuser is comparing with itself to get NLL, so we used a trajectory from diffuser as a reference.

3. Here 255 is the diffusion timestep. Dividing by 255 is not necessary. We just get the average NLL in the diffusion process in this case.
